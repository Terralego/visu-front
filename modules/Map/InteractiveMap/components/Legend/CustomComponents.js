import React from 'react';

import Circle from './components/Circle';
import Rect from './components/Rect';

const Square = ({ color, size = 10 }) => (
  <Rect color={color} size={+size} />
);

const CircleComponent = ({ color, size = 10 }) => (
  <Circle color={color} size={+size} />
);

export default [{
  tagName: 'square',
  component: Square,
  autoClose: true,
}, {
  tagName: 'circle',
  component: CircleComponent,
  autoClose: true,
}];
