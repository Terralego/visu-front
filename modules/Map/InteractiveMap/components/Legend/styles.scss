.tf-legend {
  box-shadow: 0 1px 3px 0 rgba(16,22,26,0.40);
  border-radius: 2px;
  background: #F5F8FA;
  padding: .5rem;
  margin-top: 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;

  > .tf-legend {
    border: 0;
  }

  &__title {
    margin: 0.2rem 0.3rem;
    font-size: 1rem;
    font-weight: 600;
  }

  &__list {
    display: table;
    margin: 0 .5rem;
  }



  &__symbol-container {
    display: table-cell;
    text-align: center;
    vertical-align: middle;
  }
  &__symbol {
    border-radius: 0%;
    display: inline-block;
  }
  &__label {
    font-size: 0.8rem;
    padding: 0 0 0 .5rem;
    display: table-cell;
    vertical-align: middle;
    max-width: 170px;
  }

  &__item {
    &--circle {
      display: table-row;
      .tf-legend__symbol {
        border-radius: 50%;
        display: inline-block;
      }
    }
  }
}