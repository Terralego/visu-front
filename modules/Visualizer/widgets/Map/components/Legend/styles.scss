.tf-legend {
  border: 1px solid #333;
  box-shadow: 0 0 2px rgba(0,0,0,.4);
  border-radius: 2px;
  background: white;

  &__title {
    margin: .2rem;
    font-size: 1rem;
  }

  .item {
    margin: .4rem;
    display: flex;

    &__content {
      padding: 0 1rem;
      flex: 1;
    }

    &__before,
    &__after {
      display: inline-block;
      width: 1rem;
      height: 1rem;
    }

    &__before {
      margin-right: 1rem;
    }

    &__after {
      display: none;
      margin-left: 1rem;
    }
  }
}
