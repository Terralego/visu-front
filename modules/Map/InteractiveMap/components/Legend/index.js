import Legend from './Legend';

export * from './Legend';
export default Legend;
