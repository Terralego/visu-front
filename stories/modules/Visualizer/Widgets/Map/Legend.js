import React from 'react';
import { select } from '@storybook/addon-knobs';

import Legend from '../../../../../modules/Map/InteractiveMap/components/Legend';

const items = shape => ([{
  label: 'Rouge',
  color: 'red',
  shape,
}, {
  label: 'Vert',
  color: 'green',
  shape,
}, {
  label: 'Bleu',
  color: 'blue',
  shape,
}]);

export default () => (
  <Legend
    title="Exemple de lÃ©gende"
    items={items(select(
      'Shape',
      ['square', 'circle'],
      'square',
    ))}
  />
);
