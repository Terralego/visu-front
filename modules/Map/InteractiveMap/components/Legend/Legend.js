import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

export const Legend = props => {
  const { title, items, level, position } = props;
  const biggerRadius = items.filter((int, item) => Math.max(int, item.radius), 0);

  return (
    <div className={`tf-legend tf-legend--level-${level}`}>
      <h4
        className="tf-legend__title"
      >
        {title}
      </h4>
      <div className="tf-legend__list">
        {items.map(({ label, color, items: subItems, shape = 'square', radius }) => (
          subItems
            ? (
              <Legend
                key={`${label}${items.length}`}
                title={label}
                items={subItems}
                level={level + 1}
              />
            ) : (
              <div
                key={`${label}${color}`}
                className={`tf-legend__item item--${shape} item--${position}`}
              >
                <div
                  className="item__symbol-container"
                  style={{ height: `${biggerRadius}px`, margin: '0.5rem' }}
                >
                  <div
                    className="item__symbol"
                    style={{ backgroundColor: color, width: radius, height: radius }}
                  />
                </div>
                <div
                  className="item__label"
                >
                  {label}
                </div>
              </div>
            )))}
      </div>
    </div>

  );
};

Legend.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    color: PropTypes.string,
    items: PropTypes.array,
    shape: PropTypes.string,
    radius: PropTypes.number,
  })).isRequired,
  position: PropTypes.string,
  level: PropTypes.number,
};
Legend.defaultProps = {
  position: 'left',
  level: 1,
};

export default Legend;
