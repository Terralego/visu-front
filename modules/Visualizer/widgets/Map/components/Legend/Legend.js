import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

export const Legend = ({ title, items }) => (
  <div className="tf-legend">
    <h4
      className="tf-legend__title"
    >
      {title}
    </h4>
    {items.map(({ label, color, shape = 'square' }) => (
      <div
        key={`${label}${color}`}
        className={`tf-legend__item item item--${shape}`}
      >
        <div
          className="item__before"
          style={{ backgroundColor: color }}
        />
        <div className="item__label">
          {label}
        </div>
        <div
          className="item__after"
          style={{ backgroundColor: color }}
        />
      </div>
    ))}
  </div>
);

Legend.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    shape: PropTypes.string,
  })).isRequired,
};

export default Legend;
