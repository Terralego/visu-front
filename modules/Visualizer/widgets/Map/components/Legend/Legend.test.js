import React from 'react';
import renderer from 'react-test-renderer';

import Legend from './Legend';

describe('should render correctly', () => {
  const items = shape => ([{
    label: 'Rouge',
    color: 'red',
    shape,
  }, {
    label: 'Vert',
    color: 'green',
    shape,
  }, {
    label: 'Bleu',
    color: 'blue',
    shape,
  }]);
  it('square', () => {
    const tree = renderer.create(<Legend title="Hello World" items={items()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('circle', () => {
    const tree = renderer.create(<Legend title="Hello World" items={items('circle')} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
