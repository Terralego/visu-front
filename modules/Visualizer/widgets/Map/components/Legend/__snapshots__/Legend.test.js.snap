// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly circle 1`] = `
<div
  className="tf-legend"
>
  <h4
    className="tf-legend__title"
  >
    Hello World
  </h4>
  <div
    className="tf-legend__item item item--circle"
  >
    <div
      className="item__before"
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    />
    <div
      className="item__label"
    >
      Rouge
    </div>
    <div
      className="item__after"
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    />
  </div>
  <div
    className="tf-legend__item item item--circle"
  >
    <div
      className="item__before"
      style={
        Object {
          "backgroundColor": "green",
        }
      }
    />
    <div
      className="item__label"
    >
      Vert
    </div>
    <div
      className="item__after"
      style={
        Object {
          "backgroundColor": "green",
        }
      }
    />
  </div>
  <div
    className="tf-legend__item item item--circle"
  >
    <div
      className="item__before"
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    />
    <div
      className="item__label"
    >
      Bleu
    </div>
    <div
      className="item__after"
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    />
  </div>
</div>
`;

exports[`should render correctly square 1`] = `
<div
  className="tf-legend"
>
  <h4
    className="tf-legend__title"
  >
    Hello World
  </h4>
  <div
    className="tf-legend__item item item--square"
  >
    <div
      className="item__before"
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    />
    <div
      className="item__label"
    >
      Rouge
    </div>
    <div
      className="item__after"
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    />
  </div>
  <div
    className="tf-legend__item item item--square"
  >
    <div
      className="item__before"
      style={
        Object {
          "backgroundColor": "green",
        }
      }
    />
    <div
      className="item__label"
    >
      Vert
    </div>
    <div
      className="item__after"
      style={
        Object {
          "backgroundColor": "green",
        }
      }
    />
  </div>
  <div
    className="tf-legend__item item item--square"
  >
    <div
      className="item__before"
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    />
    <div
      className="item__label"
    >
      Bleu
    </div>
    <div
      className="item__after"
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    />
  </div>
</div>
`;
